patches/server/1056-Added-EntityDamageDoorEvent-and-implemented-into-Ser.patch

```diff
@@ -0,0 +1,31 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: CMarcoo <cmarco.org@gmail.com>
+Date: Thu, 14 Mar 2024 17:04:43 +0100
+Subject: [PATCH] Added EntityDamageDoorEvent and implemented into Server.
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java b/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
+index a85885ee51df585fa11ae9f8fcd67ff2a71c5a18..f747753868274f8582792ecfd7c00a8c4cf8ae29 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
++++ b/src/main/java/net/minecraft/world/entity/ai/goal/BreakDoorGoal.java
+@@ -55,7 +55,19 @@ public class BreakDoorGoal extends DoorInteractGoal {
+     @Override
+     public void tick() {
+         super.tick();
+-        if (this.mob.getRandom().nextInt(20) == 0) {
++        // Paper start - damage door event
++        boolean playEffect = this.mob.getRandom().nextInt(20) == 0;
++        io.papermc.paper.event.entity.EntityDamageDoorEvent event = new io.papermc.paper.event.entity.EntityDamageDoorEvent(this.mob.getBukkitEntity(), this.mob.level().getBlockState(this.doorPos).createCraftBlockData(), this.breakTime, this.doorBreakTime, playEffect);
++        if (event.callEvent()) {
```

Comment:
This is wrong and break the vanilla behavior like the event was always cancelled